### DISCUSSION TITLE
Is there possible to use keyboard in the simulation to manipulate an object?

### DISCUSSION THREAD

**Discussion by EmilyMcIver** [InteractiveComputerGraphics/PositionBasedDynamics#144] [Q&A]
Is there possible to use keyboard in the simulation to manipulate an object?
Can you explain how to do this?
And do you have the working link where is the scene file format is explain?  https://positionbaseddynamics.readthedocs.io/en/latest/file_format

**Comment by janbender** [comment]
You can implement this in the following way:

// Call this in the beginning of your simulation:
MiniGL::addKeyFunc('a', action);

// callback function that is called when key 'a' is pressed
void action()
{ 
  doSomething();
}

Sorry but the file format is not documented yet.

**Reply by EmilyMcIver** [reply]
Do you know how to use keyboard to manipulate an object in the https://github.com/InteractiveComputerGraphics/SPlisHSPlasH library? Because there we use .json file to describe the scene.

**Comment by janbender** [comment]
You can do it in the same way. Just implement an action function. Alternatively you can use the Python version and implement the key actions in Python.
